hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -358,23 +364,27 @@ public static String createNewFileId(String idPfx, int id) {
    * Get the file extension from the log file.
    */
   public static String getFileExtensionFromLog(Path logPath) {
-    Matcher matcher = LOG_FILE_PATTERN.matcher(logPath.getName());
+    boolean isArchivedLog = logPath.getName().contains(ARCHIVED_LOG_PREFIX);
+    Matcher matcher =  isArchivedLog ? ARCHIVED_LOG_FILE_PATTERN.matcher(logPath.getName()) :
+        LOG_FILE_PATTERN.matcher(logPath.getName());
     if (!matcher.find()) {
       throw new InvalidHoodiePathException(logPath, "LogFile");
     }
-    return matcher.group(3);
+    return isArchivedLog ? ARCHIVE_STR : matcher.group(3);
```

Comment:
Any reason to handle archived log separately?  Can't the `LOG_FILE_PATTERN` be adjusted using OR operand (`log|archive`) to match `.log` or `.archive`?